packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -406,18 +555,23 @@ export class DataStores implements IDisposable {
 		const context = new LocalDetachedFluidDataStoreContext({
 			id,
 			pkg,
-			runtime: this.runtime,
-			storage: this.runtime.storage,
-			scope: this.runtime.scope,
-			createSummarizerNodeFn: this.getCreateChildSummarizerNodeFn(id, {
+			parentContext: this.createParentContext(id),
+			storage: this.parentContext.storage,
+			scope: this.parentContext.scope,
+			createSummarizerNodeFn: this.parentContext.getCreateChildSummarizerNodeFn(id, {
 				type: CreateSummarizerNodeSource.Local,
 			}),
 			makeLocallyVisibleFn: () => this.makeDataStoreLocallyVisible(id),
 			snapshotTree: undefined,
 			isRootDataStore: isRoot,
 			loadingGroupId,
 			channelToDataStoreFn: (channel: IFluidDataStoreChannel, channelId: string) =>
-				channelToDataStore(channel, channelId, this.runtime, this, this.runtime.logger),
+				channelToDataStore(
```

Comment:
I wonder what sort of implications this might cause. When we attach the the `FluidDataStoreRuntime` to a `IFluidDataStoreContext`, we want to return also a `IDataStore` object. But further down the stack, we might not want any sort of aliasing. It would/should break right? I'm not exactly sure what implications we have to deal with.

Reply:
#todo figure this out in an end-to-end test. 

@tyler-cai-microsoft do you have a scenario in mind? Not sure if it's possible to bypass existing safeguards from the API. I don't believe there's any way of trying to alias a nested datastore, right? We can create them directly using `_createFluidDataStoreContext` but we don't have `channelToDataStore` exposed.

Now, we could expose a better set of APIs on the DataStores to create proper IDatastore's and we'd have two options there:

a) ensure that aliasing always works at the runtime level - for now
b) create a new interface for un-aliasible IDatastore for DataStores to return

I prefer a)

Reply:
We should definitely look deeper into layering and how our APIs are built (as responsibilities are quite often in the wrong places).
But I do not understand the initial concern of "we want to return also a IDataStore object". We control what nested APIs nested data stores have. Or rather - specific user of nested DataStores control, as there is no way to expose generic nested data stores (at least not yet). If such user decides not to expose IDataStore, then it's not exposed there. It has nothing to do with what context does.